### PR TITLE
TransactionScenario increment

### PR DESCRIPTION
**Pull request by vaclavrek**
*Description of changes:*
Creating incremental ETL (backend) for TransactionScenario

*VSTS work item:*

*Items to complete before pull request is merged:*
[ ] Reviewed and approved by Blackbaud
[ ] VSTS work item linked above
[ ] Change deployed to CDEV *(`develop` branch only)* or ATE *(`master` branch only)*
[ ] Change tested in CDEV or ATE
[ ] Change deployed to PROD *(`master` branch only)*

### CODE REVIEW

File: SQL/10_1_out_TransactionScenario.sql [hunk 1]
```diff
@@ -1,39 +1,123 @@
...
+from stg_csv_SummarizedTransaction_merge t	
+left join tmp_transaction_scenario abs
+on  T.AccountId = abs.AccountId 
+   and abs.FiscalPeriodId = T.FiscalPeriodId
+   and abs.tenantid = t.tenantid
+   and abs._sys_is_deleted = false
+where   (t.tenantid, t.TranDistributionId) in
+( select tenantid, 1000000000000 + SummaryId from stg_csv_SummarizedTransaction_merge where (_sys_updated_at > nvl(to_timestamp(nullif('${PREV_TMP_TS['out_TransactionScenario#stg_csv_SummarizedTransaction_merge']}',''),'yyyy-mm-dd hh24:mi:ss.us'),'2000-01-01') and _sys_updated_at <= to_timestamp('${LAST_TMP_TS['stg_csv_SummarizedTransaction_merge']}','yyyy-mm-dd hh24:mi:ss.us'))
+    union all
+  select tenantid, TranDistributionId from out_transactions where (tenantid,AccountId,FiscalPeriodId) in (select tenantid, AccountId, FiscalPeriodId from tmp_transaction_scenario where _sys_updated_at > nvl(to_timestamp(nullif('${PREV_TMP_TS['out_TransactionScenario#tmp_transaction_scenario']}',''),'yyyy-mm-dd hh24:mi:ss.us'),'2000-01-01') and _sys_updated_at <= to_timestamp('${LAST_TMP_TS['tmp_transaction_scenario']}','yyyy-mm-dd hh24:mi:ss.us'))
```
Comment (Blackbaud-NitishSharma) [inline SQL/10_1_out_TransactionScenario.sql]: @vaclavrek - Avoid hard code values '2000-01-01'.

File: SQL/10_1_out_TransactionScenario.sql [hunk 2]
```diff
@@ -1,39 +1,123 @@
...
+( select tenantid, TranDistributionId from out_transactions where _sys_updated_at > nvl(to_timestamp(nullif('${PREV_TMP_TS['out_TransactionScenario#out_transactions']}',''),'yyyy-mm-dd hh24:mi:ss.us'),'2000-01-01') and _sys_updated_at <= to_timestamp('${LAST_TMP_TS['out_transactions']}','yyyy-mm-dd hh24:mi:ss.us')
+   union all
+  select tenantid, TranDistributionId from out_transactions where (tenantid,AccountId,FiscalPeriodId) in (select tenantid, AccountId, FiscalPeriodId from tmp_transaction_scenario where _sys_updated_at > nvl(to_timestamp(nullif('${PREV_TMP_TS['out_TransactionScenario#tmp_transaction_scenario']}',''),'yyyy-mm-dd hh24:mi:ss.us'),'2000-01-01') and _sys_updated_at <= to_timestamp('${LAST_TMP_TS['tmp_transaction_scenario']}','yyyy-mm-dd hh24:mi:ss.us'))
+)
+;
+
+insert /*+ direct */ into wrk_out_TransactionScenario
+select
+	t.TenantId as "TenantId"
+	,1000000000000 + t.SummaryId as "TransactionDistributionId"
```
Comment (Blackbaud-NitishSharma) [inline SQL/10_1_out_TransactionScenario.sql]: @vaclavrek - Any specific significance of hard coded value (1000000000000) ?

File: DDL/out_TransactionScenario.sql [hunk 3]
```diff
@@ -1,15 +1,26 @@
+drop table if exists wrk_out_TransactionScenario;
+drop table if exists wrk_out_TransactionScenario_diff;
 drop table if exists out_TransactionScenario;
+
 CREATE TABLE out_TransactionScenario
 (
-    _sys_transform_id int NOT NULL encoding rle,
     TenantId varchar(255) encoding rle,
```
Comment (Blackbaud-NitishSharma) [inline DDL/out_TransactionScenario.sql]: @vaclavrek - is it fine to remove _sys_transform_id.

File: SQL/10_1_out_TransactionScenario.sql (hunk 1 above)
Comment (vaclavrek) [reply SQL/10_1_out_TransactionScenario.sql]: @Blackbaud-NitishSharma , what would you like to use instead? This value is used as safe "past" point in time for incremental metadata timeframe (for init loads etc).

File: SQL/10_1_out_TransactionScenario.sql (hunk 2 above)
Comment (vaclavrek) [reply SQL/10_1_out_TransactionScenario.sql]: @Blackbaud-NitishSharma , this value was used by BB to distinguish between transaction IDs and sumarized transaction IDs. I don't think it has any specific meaning.

**Pull request merged**